### PR TITLE
Remove stale OutOfDisk node condition for 1.13 clusters

### PR DESCRIPTION
**Pull request by rfranzke**
**What this PR does / why we need it**: Shoot care controller's garbage collection does now remove stale `OutOfDisk` node conditions for 1.13 Kubernetes clusters.

Further references:
- gardener/machine-controller-manager#214
- kubernetes/kubernetes#72507

**Special notes for your reviewer**:
Once a new 1.13 hotfix Kubernetes release has been cut we can limit the functionality to only clusters with versions `>= 1.13.0, < 1.13.3` (assuming `1.13.3` contains the fix).

**Release note**:

```improvement operator
The shoot care controller does now remove stale `OutOfDisk` node conditions for 1.13 clusters. See https://github.com/kubernetes/kubernetes/pull/72507 for more details.
```

### CODE REVIEW

File: pkg/operation/botanist/garbage_collection.go [hunk 1]
```diff
@@ -95,3 +102,41 @@ func (b *Botanist) deleteStalePods(client kubernetes.Interface, podList *corev1.
...
+			conditions []corev1.NodeCondition
+			nodeCopy   = node.DeepCopy()
+		)
+
+		for _, condition := range node.Status.Conditions {
+			if condition.Type != corev1.NodeOutOfDisk {
+				conditions = append(conditions, condition)
+			}
+		}
+		nodeCopy.Status.Conditions = conditions
```
Comment (adracus) [inline pkg/operation/botanist/garbage_collection.go]: Shouldn't we do a quick check if `len(conditions) != len(nodeCopy.Status.Conditions)` before we do the update? Then we can reduce no-ops.

File: pkg/operation/botanist/garbage_collection.go [hunk 2]
```diff
@@ -95,3 +102,41 @@ func (b *Botanist) deleteStalePods(client kubernetes.Interface, podList *corev1.
...
+	var nodeList *corev1.NodeList
+	if err := b.K8sShootClient.Client().List(context.TODO(), nil, nodeList); err != nil {
+		return err
+	}
+
+	var result error
+	for _, node := range nodeList.Items {
+		var (
+			conditions []corev1.NodeCondition
+			nodeCopy   = node.DeepCopy()
```
Comment (adracus) [inline pkg/operation/botanist/garbage_collection.go]: Why the copy?

**Review by adracus — CHANGES_REQUESTED** [review]

**Comment by rfranzke** [comment]
ping

File: pkg/operation/botanist/garbage_collection.go [hunk 3]
```diff
@@ -95,3 +102,43 @@ func (b *Botanist) deleteStalePods(client kubernetes.Interface, podList *corev1.
 
 	return result
 }
+
+func (b *Botanist) removeStaleOutOfDiskNodeCondition() error {
+	// TODO: This code can be limited to 1.13.0-1.13.3 (assuming 1.13.3 has contains the Kubernetes fix) as soon as
+	// a new release containing the fix has been cut.
```
Comment (mvladev) [inline pkg/operation/botanist/garbage_collection.go]: Can you add @afritzler's PR https://github.com/kubernetes/kubernetes/pull/73394 which should be merged in `1.13.3`?

Comment (rfranzke) [reply pkg/operation/botanist/garbage_collection.go]: done

**Review by timuthy — APPROVED** [review]
/lgtm

**Pull request merged**